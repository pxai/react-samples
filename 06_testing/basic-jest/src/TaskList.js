import React, { Component } from "react";
import Task from "./Task";
import Form from "./Form";

class TaskList extends Component {
    constructor() {
        super();
        this.state = { showForm: false, tasks: [] };
    }

    componentDidMount() {
        this.getTasks();
    }

    render() {
        return <div>
            <h3>Tasks: { this.state.tasks.length}</h3>
            {
                this.state.tasks.map(task =>
                    <Task key={task.id} task={task}
                          onRemove={this.remove.bind(this)}
                          onUpdate={this.update.bind(this)} />
                )
            }
            <div><a href="javascript:void(0)" onClick={this.toggleForm.bind(this)}>Add Task</a></div>
            {this.state.showForm && <Form onUpdate={this.add.bind(this)} />}
        </div>;
    }

    getTasks() {
        const tasks = this.props.api.findAll();
        this.setState({ tasks });
    }

    add({ name }) {
        this.props.api.add({ name });
        this.toggleForm();
        this.getTasks();
    }

    update(task) {
        this.props.api.update(task);
        this.getTasks();
    }


    remove(id) {
        this.props.api.remove(id);
        this.getTasks();
    }

    toggleForm() {
        this.setState({ showForm: !this.state.showForm });
    }
}

export default TaskList;
