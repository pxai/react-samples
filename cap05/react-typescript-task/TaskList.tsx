import * as React from 'react';
import Task from "./Task";
import Form from "./Form";
import Api from "./Api";
import TaskComponent from "./TaskComponent";

interface TaskListProps {
    api: Api 
}

interface TaskListState {
    showForm: boolean
    tasks: Task[]
}

class TaskList extends React.Component<TaskListProps, TaskListState> {
    public constructor (props: TaskListProps) {
        super(props);

        this.state = { showForm: false, tasks: [] };
    }

    public render (): React.ReactNode {
        return <div>
            <h3>Tasks: { this.state.tasks.length}</h3>
            {
                this.state.tasks.map(task =>
                    <TaskComponent 
                        key={task.id} 
                        task={task} 
                        onRemove={this.remove.bind(this)} onUpdate={this.update.bind(this)} />
                )
            }
            <div><a href="javascript:void(0)" onClick={this.toggleForm.bind(this)}>Add Task</a></div>
            {this.state.showForm && <Form onUpdate={this.add.bind(this)} />}
        </div>;
    }

    private getTasks () {
        const tasks = this.props.api.findAll();
        this.setState({ tasks });
    }

    private add({ name }: Task) {
        this.props.api.add({ name } as Task);
        this.toggleForm();
        this.getTasks();
    }

    private update(task) {
        this.props.api.update(task);
        this.getTasks();
    }


    private remove(id) {
        this.props.api.remove(id);
        this.getTasks();
    }

    private toggleForm() {
        this.setState({ showForm: !this.state.showForm });
    }
}

export default TaskList;
