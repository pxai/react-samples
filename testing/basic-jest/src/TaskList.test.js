import React from 'react';
import ReactDOM from 'react-dom';
import TaskList from './TaskList';
import renderer from 'react-test-renderer';

test('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<TaskList />, div);
  ReactDOM.unmountComponentAtNode(div);
});


test('it matches snapshot', () => {
	const component = renderer.create(<TaskList />);
 let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

