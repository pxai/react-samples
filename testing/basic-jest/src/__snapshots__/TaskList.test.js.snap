// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it matches snapshot 1`] = `
<div>
  <h3>
    Tasks: 
    2
  </h3>
  <div>
    <div>
      2
      - 
    </div>
    <div>
      <a
        href="javascript:void(0)"
        onClick={[Function]}
      >
        Update
      </a>
       |
      <a
        href="javascript:void(0)"
        onClick={[Function]}
      >
        Delete
      </a>
    </div>
  </div>
  <div>
    <div>
      4
      - 
    </div>
    <div>
      <a
        href="javascript:void(0)"
        onClick={[Function]}
      >
        Update
      </a>
       |
      <a
        href="javascript:void(0)"
        onClick={[Function]}
      >
        Delete
      </a>
    </div>
  </div>
  <div>
    <a
      href="javascript:void(0)"
      onClick={[Function]}
    >
      Add Task
    </a>
  </div>
</div>
`;
